snapshotter/demux/cache/evict.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/firecracker-microvm/firecracker-containerd/snapshotter/demux/proxy"
+)
+
+// EvictionPolicy defines the interface for enforcing a cache eviction policy.
+type EvictionPolicy interface {
+	Enforce(key string)
+}
+
+type evictionPolicy struct {
+	evict chan string
+	stop  chan struct{}
+}
+
+// EvictOnConnectionFailurePolicy defines an eviction policy where entries are evicted
+// from cache after a failed connection attempt occurs.
+type EvictOnConnectionFailurePolicy struct {
+	evictionPolicy
+
+	dialer proxy.Dialer
+
+	frequency time.Duration
+}
+
+// NewEvictOnConnectionFailurePolicy creates a new instance with
```

Comment:
nit: godoc not complete

Reply:
Great catch. Let's resolve this. I pushed a new revision with the doc reworked.